### PR TITLE
Add csi nodeexpandsecret blog article

### PR DESCRIPTION
**Pull request by humblec**
Ref# KEP: https://github.com/kubernetes/enhancements/pull/3173/
Implementation: https://github.com/kubernetes/kubernetes/pull/105963
Doc : https://github.com/kubernetes/we

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
Cc @xing-yang

File: content/en/blog/_posts/2022-04-19-csi-nodeexpandsecret.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+layout: blog
+title: "Kubernetes 1.24: Use secrets while expanding csi volumes on Node"
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-19-csi-nodeexpandsecret.md:3]: early nit:
```suggestion
title: "Kubernetes 1.24: Use Secrets While Expanding CSI Volumes On Nodes"
```

Maybe there is a better title? I might suggest a new one once I've seen what the article is introducing.

Comment (humblec) [reply content/en/blog/_posts/2022-04-19-csi-nodeexpandsecret.md:3]: @sftim sure, I was also in a  dilemma about the subject, so inputs are welcome!  :+1:

**Comment by sftim** [comment]
@humblec if this is ready for review, please edit the PR title to remove the `[WIP] `  prefix

**Comment by humblec** [comment]
> @humblec if this is ready for review, please edit the PR title to remove the `[WIP] ` prefix

The feature is at last stage of review, thats why I kept the `WIP` in the title, however if that does not matter, this is good for review @sftim . I have removed the WIP from the title :+1:

File: content/en/blog/_posts/2022-04-19-csi-nodeexpandsecret.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+configuration as part of NodeExpansion  by the CSI drivers thus make use of
+the same to perform node side expansion operation with the underlying
+storage system.
+
+## What is this all about?
+
+
+Till this version  of kubernetes , the CSI drivers didnt have a mechanism to
+make use of secretRef at time of Node operation (ex: nodeExpansion) as the
+subjected csi request does not carry a secret or credentials in the request.
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-19-csi-nodeexpandsecret.md:22]:
```suggestion
Before Kubernetes v1.24, you were able to define a cluster-level StorageClass
that made use of [secrets for volume access](https://kubernetes-csi.github.io/docs/secrets-and-credentials-storage-class.html), but you didn't have any mechanism to specify the
secret that was used for operations that take place when the storage was
mounted onto a node.
```

File: content/en/blog/_posts/2022-04-19-csi-nodeexpandsecret.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+```
+
+## The future
+
+As this feature is still in alpha, we expect to update or get feedback from CSI driver
+authors with more tests and implementation. The community plans to eventually
+promote the feature to Beta in upcoming releases.
+
+## How can I learn more?
+
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-19-csi-nodeexpandsecret.md:167]: Please also link to the docs on setting secrets for a StorageClass / PersistentVolume.

**Review by sftim — CHANGES_REQUESTED** [review]
Hi, thanks.

Overall, I think this will work better if you make it a less technical announcement that links to docs about setting and using these secrets.
What's the URL of the documentation change that we made for this enhancement? I'd like to read that for context.

File: content/en/blog/_posts/2022-04-19-csi-nodeexpandsecret.md (hunk 3 above)
Comment (humblec) [reply content/en/blog/_posts/2022-04-19-csi-nodeexpandsecret.md:175]: Done

**Comment by humblec** [comment]
> Hi, thanks.
> 
> Overall, I think this will work better if you make it a less technical announcement that links to docs about setting and using these secrets. What's the URL of the documentation change that we made for this enhancement? I'd like to read that for context.

https://github.com/kubernetes/website/pull/32298 covers doc part of it.  Thanks @sftim

**Comment by humblec** [comment]
> @humblec I recommend that you keep this text ready for a future branch, but close this PR.

Sure, and thanks for all your review comments ! Will continue or take this version to next attempt. Thanks @sftim

**Pull request closed without merging**